plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObjectPdo.class.php

```diff
@@ -612,32 +619,80 @@ public function getNameAccessPoints()
   }
 
   /*
-   * Get related terms and its ancestors
+   * Get related terms plus any ancestors
    */
   protected function getRelatedTerms($typeId)
   {
+    $relatedTerms = array();
+
     $sql  = 'SELECT
-                DISTINCT term.id,
-                term.taxonomy_id,
-                term.source_culture,
-                slug.slug,
-                i18n.name';
+                DISTINCT current.id';
     $sql .= ' FROM '.QubitObjectTermRelation::TABLE_NAME.' otr';
     $sql .= ' JOIN '.QubitTerm::TABLE_NAME.' current
                 ON otr.term_id = current.id';
-    $sql .= ' JOIN '.QubitTerm::TABLE_NAME.' term
-                ON term.lft <= current.lft AND term.rgt >= current.rgt';
-    $sql .= ' JOIN '.QubitTermI18n::TABLE_NAME.' i18n
-                ON term.id = i18n.id';
-    $sql .= ' JOIN '.QubitSlug::TABLE_NAME.' slug
-                ON term.id = slug.object_id';
     $sql .= ' WHERE otr.object_id = ?
-               AND term.taxonomy_id = ?';
+                AND current.taxonomy_id = ?';
 
     self::$statements['relatedTerms'] = self::$conn->prepare($sql);
     self::$statements['relatedTerms']->execute(array($this->__get('id'), $typeId));
 
-    return self::$statements['relatedTerms']->fetchAll(PDO::FETCH_OBJ);
+    // Get directly related terms.
+    $rows = self::$statements['relatedTerms']->fetchAll(PDO::FETCH_ASSOC);
+
+    if (0 == count($rows))
+    {
+      return $relatedTerms;
+    }
+
+    // Iterate over each directly related term, adding all ancestors of each
+    foreach($rows as $row)
+    {
+      $relatedTerms = array_merge($relatedTerms, $this->recursivelyGetParentTerms(array($row['id'])));
+    }
+
+    // Merge unique related terms to $rows
+    $relatedTerms = array_unique($relatedTerms);
+    sort($relatedTerms, SORT_NUMERIC);
```

Comment:
Does sorting really matters? It may saves us some extra milliseconds.

Reply:
I have removed this line.